src-tauri/src/lib.rs

```diff
@@ -48,19 +46,32 @@ async fn get_swap_infos_all(
         .to_string_result()
 }
 
-/*macro_rules! tauri_command {
-    ($command_name:ident, $command_args:ident, $command_response:ident) => {
-        #[tauri::command]
-        async fn $command_name(
-            context: State<'_, Context>,
-            args: $command_args,
-        ) -> Result<$command_response, String> {
-            swap::api::request::$command_name(args, context)
-                .await
-                .to_string_result()
-        }
-    };
-}*/
+#[tauri::command]
+async fn buy_xmr(
+    context: State<'_, Arc<Context>>,
+    args: BuyXmrArgs,
+) -> Result<BuyXmrResponse, String> {
+    buy_xmr_impl(args, context.inner().clone())
+        .await
+        .to_string_result()
+}
+
+#[tauri::command]
+async fn get_history(context: State<'_, Arc<Context>>) -> Result<GetHistoryResponse, String> {
+    get_history_impl(context.inner().clone())
+        .await
+        .to_string_result()
+}
+
+#[tauri::command]
+async fn resume_swap(
+    context: State<'_, Arc<Context>>,
+    args: ResumeSwapArgs,
+) -> Result<ResumeSwapResponse, String> {
+    resume_swap_impl(args, context.inner().clone())
+        .await
+        .to_string_result()
```

Comment:
We should be using the new Request trait and the macro by @Einliterflasche here. Somehow got lost during rebase I suppose